exporter/fileexporter/file_exporter.go

```diff
@@ -93,6 +113,20 @@ func exportMessageAsLine(e *fileExporter, buf []byte) error {
 	return nil
 }
 
+func exportMessageAsBuffer(e *fileExporter, buf []byte) error {
+	// Ensure only one write operation happens at a time.
+	e.mutex.Lock()
+	defer e.mutex.Unlock()
+	// write the size of each message before writing the message itself.
+	if _, err := e.file.Write([]byte(cast.ToString(len(buf)) + "\n")); err != nil {
```

Comment:
Is the "cast" library being used as a performance optimization? 

I believe `strconv.Itoa` should be used unless there is a good reason otherwise. It's a standard library but also is more performant than `fmt.Sprintf`.